From a66b4ddfb4399cdc03ccc6f66acd634c692ee88f Mon Sep 17 00:00:00 2001
From: Igor Gnatenko <ignatenkobrain@fedoraproject.org>
Date: Apr 17 2019 05:38:03 +0000
Subject: kojid: Download only 'repomd.xml'


We have pretty slow connection from s390x koji which helped to uncover
this part. Kojid downloads all files from repomd.xml (incl. filelists)
which is really big. What we really want is just find location of
'origin' (used later in the code).

Signed-off-by: Igor Gnatenko <ignatenkobrain@fedoraproject.org>

---

diff --git a/builder/kojid b/builder/kojid
index 75f8a7d..5110d04 100755
--- a/builder/kojid
+++ b/builder/kojid
@@ -739,6 +739,9 @@ class BuildRoot(object):
             h.setopt(librepo.LRO_REPOTYPE, librepo.LR_YUMREPO)
             h.setopt(librepo.LRO_URLS, [repo_url])
             h.setopt(librepo.LRO_DESTDIR, tmpdir)
+            # We are using this just to find out location of 'origin',
+            # we don't even need to download it since we use openRemoteFile
+            h.setopt(librepo.LRO_YUMDLIST, [])
             h.perform(r)
             pkgorigins = r.getinfo(librepo.LRR_YUM_REPOMD)['origin']['location_href']
             koji.util.rmtree(tmpdir)

